tests/test-generators.py

```diff
@@ -143,29 +144,56 @@ def validate_except_cost(output1: str, output2: str) -> None:
         MEMPOOL_MODE | flags,
         version=2,
     )
+
+    stdout.write(f"{name} running generator2 (costed)...\r")
+    stdout.flush()
+    costed2 = run_generator(
+        g,
+        COST_SHATREE | flags,
+        version=2,
+    )
+
     if run_generator1:
         validate_except_cost(mempool.output, mempool2.output)
 
     with open(g) as f:
         expected = f.read().split("\n", 1)[1]
-        if not "STRICT" in expected:
-            expected_mempool = expected
-            if (
-                consensus2.result is not None
-                and mempool2.result is not None
-                and consensus2.result.cost != mempool2.result.cost
-            ):
-                print("\n\ncost when running in mempool mode differs from normal mode!")
-                failed = 1
+        if "STRICT:\n" in expected:
+            # split STRICT section
+            consensus_part, rest = expected.split("STRICT:\n", 1)
+
+            if "COSTED_SHA:\n" in rest:
```

Comment:
I have the same comment here. The test cases become much more difficult to read by duplicating the outputs like this. I really think you should just have separate outputs for the cost